staging/src/k8s.io/kubectl/pkg/drain/drain.go

```diff
@@ -274,8 +282,14 @@ func (d *Helper) evictPods(pods []corev1.Pod, evictionGroupVersion schema.GroupV
 			for {
 				switch d.DryRunStrategy {
 				case cmdutil.DryRunServer:
+					if d.OnPodDeletionOrEvictionStarted != nil {
```

Comment:
Are you sure you want to have the hook invoked even in dry-run mode? Even if it's a server one.

Reply:
other hooks are not invoked in DryRunServer, I will remove this one too.